examples/MenuExample.swift

```diff
@@ -0,0 +1,90 @@
+/*
+ Copyright 2017-present The Material Motion Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import UIKit
+import Transitioning
+
+// This example demonstrates the minimal path to building a custom transition using the Material
+// Motion Transitioning APIs in Swift. The essential steps have been documented below.
+
+class MenuExampleViewController: ExampleViewController {
+
+  func didTap() {
+    let modalViewController = ModalViewController()
+    modalViewController.transitionController.transition = MenuTransition()
+    present(modalViewController, animated: true)
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    let label = UILabel(frame: view.bounds)
+    label.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+    label.textColor = .white
+    label.textAlignment = .center
+    label.text = "Tap to start the transition"
+    view.addSubview(label)
+
+    let tap = UITapGestureRecognizer(target: self, action: #selector(didTap))
+    view.addGestureRecognizer(tap)
+  }
+
+  override func exampleInformation() -> ExampleInfo {
+    return .init(title: type(of: self).catalogBreadcrumbs().last!,
+                 instructions: "Tap to present a modal transition.")
+  }
+}
+
+private final class MenuTransition: NSObject, TransitionWithPresentation {
+  func presentationController(forPresented presented: UIViewController, presenting: UIViewController, source: UIViewController?) -> UIPresentationController? {
+    return nil
+  }
+
+  func defaultModalPresentationStyle() -> UIModalPresentationStyle {
+    return .overCurrentContext
+  }
+  
+  func start(with context: TransitionContext) {
+    let backVC = context.backViewController
+    let foreVC = context.foreViewController
+    let containerView = context.containerView
+    
+    if(context.direction == .forward) {
+      containerView.insertSubview(foreVC.view, aboveSubview: backVC.view)
```

Comment:
You don't need to add any views to the view hierarchy, `Transition` does that for you.